pkg/vault/contracts/BatchRouter.sol

```diff
@@ -671,6 +671,317 @@ contract BatchRouter is IBatchRouter, BatchRouterStorage, RouterCommon, Reentran
+                    BatchRouter.addLiquidityBoostedPoolUnbalancedHook.selector,
+                    AddLiquidityHookParams({
+                        sender: msg.sender,
+                        pool: pool,
+                        maxAmountsIn: exactUnderlyingAmountsIn,
+                        minBptAmountOut: 0,
+                        kind: AddLiquidityKind.UNBALANCED,
+                        wethIsEth: false,
+                        userData: userData
+                    })
+                )
+            ),
+            (uint256[], uint256, bytes)
+        );
+    }
+
+    /// @inheritdoc IBatchRouter
+    function queryAddLiquidityProportionalToBoostedPool(
+        address pool,
+        uint256[] memory maxUnderlyingAmountsIn,
+        uint256 exactBptAmountOut,
+        bytes memory userData
+    ) external saveSender returns (uint256[] memory amountsIn) {
+        (amountsIn, , ) = abi.decode(
+            _vault.quote(
+                abi.encodeWithSelector(
+                    BatchRouter.addLiquidityBoostedPoolProportionalHook.selector,
+                    AddLiquidityHookParams({
+                        sender: msg.sender,
+                        pool: pool,
+                        maxAmountsIn: maxUnderlyingAmountsIn,
+                        minBptAmountOut: exactBptAmountOut,
+                        kind: AddLiquidityKind.PROPORTIONAL,
+                        wethIsEth: false,
+                        userData: userData
+                    })
+                )
+            ),
+            (uint256[], uint256, bytes)
+        );
+    }
+
+    /// @inheritdoc IBatchRouter
+    function queryRemoveLiquidityProportionalToBoostedPool(
+        address pool,
+        uint256 exactBptAmountIn,
+        bytes memory userData
+    ) external saveSender returns (uint256[] memory amountsOut) {
+        (, amountsOut, ) = abi.decode(
+            _vault.quote(
+                abi.encodeWithSelector(
+                    BatchRouter.removeLiquidityBoostedPoolProportionalHook.selector,
+                    RemoveLiquidityHookParams({
+                        sender: msg.sender,
+                        pool: pool,
+                        minAmountsOut: new uint256[](2),
+                        maxBptAmountIn: exactBptAmountIn,
+                        kind: RemoveLiquidityKind.PROPORTIONAL,
+                        wethIsEth: false,
+                        userData: userData
+                    })
+                )
+            ),
+            (uint256, uint256[], bytes)
+        );
+    }
+
+    function addLiquidityBoostedPoolUnbalancedHook(
+        AddLiquidityHookParams calldata params
+    )
+        external
+        nonReentrant
+        onlyVault
+        returns (uint256[] memory amountsIn, uint256 bptAmountOut, bytes memory returnData)
+    {
+        IERC20[] memory boostedPoolTokens = _vault.getPoolTokens(params.pool);
+        uint256[] memory wrappedAmounts = _wrapToken(
```

Comment:
I didn't quite get this point. what argument do you mean?

Reply:
You could drop the final `uint256 limitRaw` parameter from `_wrapToken`, and inside `_wrapToken`, say:

uint256 limitRaw = kind == SwapKind.EXACT_IN ? 0 : type(uint256).max;

Reply:
yeah I agree; we can get the limit inside `_wrap`.

Also, there's a constant defined for the max limit, which is `uint128.max`. This is because the limit is scaled inside the vault, and for tokens with less than 18 decimals `uint256.max` will revert when doing so (check batch router for reference).